reana_commons/k8s/volumes.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2019 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""REANA Kubernetes volumes."""
+
+import os
+
+import json
+
+from reana_commons.config import REANA_STORAGE_BACKEND
+
+REANA_SHARED_VOLUME_NAME = "reana-shared-volume"
+REANA_CEPHFS_CLAIM_NAME = "manila-cephfs-pvc"
+
+
+def get_k8s_cephfs_volume():
+    """Return k8s CephFS volume template.
+
+    :returns: k8s CephFS volume spec as a dictionary.
+    """
+    return {
+        "name": REANA_SHARED_VOLUME_NAME,
+        "persistentVolumeClaim": {
+            "claimName": REANA_CEPHFS_CLAIM_NAME
+        },
+        "readOnly": "false"
+    }
+
+
+def get_k8s_cvmfs_volume(repository):
+    """Render k8s CVMFS volume template.
+
+    :param repository: CVMFS repository to be mounted.
+    :returns: k8s CVMFS volume spec as a dictionary.
+    """
+    return {
+        "name": "{}-cvmfs-volume".format(repository),
+        "persistentVolumeClaim": {
+            "claimName": "csi-cvmfs-{}-pvc".format(repository)
+        },
+        "readOnly": "true"
+    }
+
+
+def get_k8s_hostpath_volume(root_path):
+    """Render k8s HostPath volume template.
+
+    :param root_path: Root path in the host machine to be mounted.
+    :returns: k8s HostPath spec as a dictionary.
+    """
+    return {
+        "name": REANA_SHARED_VOLUME_NAME,
+        "hostPath": {
+            "path": root_path
+        }
+    }
+
+
+def get_shared_volume(workflow_workspace, shared_volume_root):
+    """Get shared CephFS/hostPath volume to a given job spec.
```

Comment:
I am not sure if it should be `to a given job spec` or `to a given job or workflow` ?